src/Analysis/Ast/Impl/Specializations/BuiltinsSpecializations.cs

```diff
@@ -91,6 +93,79 @@ public static IMember List(IPythonInterpreter interpreter, IPythonFunctionOverlo
             return args.Count > 0 && args[0] is PythonCollection c ? c.Contents.FirstOrDefault() : null;
         }
 
+        public static IMember Super(IPythonModule declaringModule, IPythonFunctionOverload overload, IArgumentSet argSet, IndexSpan indexSpan) {
+            if (argSet.Arguments.Any()) {
+                // If multiple arguments first argument is required
+                var classType = argSet.Argument<IMember>(0).GetPythonType<IPythonClassType>();
+                if (classType?.Mro == null) {
+                    return null;
+                }
+
+                // second argument optional
+                bool isUnbound = argSet.Argument<IMember>(1) == null;
+                if (isUnbound) {
+                    return CreateSuper(argSet, classType.Mro);
+                } else {
+                    // Second argument can be either an object instance or a class which are both IPythonType
+                    IPythonType objOrType = null;
+                    objOrType = argSet.Argument<IMember>(1).GetPythonType();
+                    bool isInstance = objOrType?.Equals(classType) ?? false;
```

Comment:
is there a helper function for isInstance ?

Reply:
`is IPythonInstance`. But do you care? `IVariable` can also be used I think, i.e.
```python
class A: ...
x = A()
class B(x): ...
```
`x` will be `IVariable` with value of `IPythonInstance` which has type of `IPythonClassType`.

Reply:
I added IsInstanceOf(..)